File: arsh_goyal_sde_sheet/arrays/75. Sort Colors/bubble_sort.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        for(int i = 0; i < nums.size(); i++) {
            bool is_sorted = true;
            for(int j = 0; j < nums.size() - i - 1; j ++) {
                if(nums[j] > nums[j+1]) {
                    swap(nums[j], nums[j+1]);
                    is_sorted = false;
                }
            }
            if(is_sorted) break;
        }
    }
};

File: arsh_goyal_sde_sheet/arrays/75. Sort Colors/bubble_sort.py
class Solution:
    # bubble sort
    def sortColors(self, nums: List[int]) -> None:
        N = len(nums)
        for i in range(N):
            already_sorted = True
            for j in range(N-i-1):
                if nums[j] > nums[j+1]:
                    already_sorted = False
                    nums[j], nums[j+1] = nums[j+1], nums[j]
            
            # slight optimisation
            if already_sorted: break
        
        